projects/packages/waf/tests/php/unit/WafRuntimeTargetsTest.php

```diff
@@ -210,144 +214,162 @@ public function provideArrayTargetsNames() {
 	/**
 	 * Provide data to test key/value targets such as REQUEST_HEADERS, ARGS, TX, etc.
 	 */
-	public function provideArrayTargets() {
+	public static function provideArrayTargets() {
 		// REQUEST_HEADERS
-		$expected = array(
+		$expected       = array(
 			array( 'header-a', 'testa' ),
 			array( 'header-b', 'testb' ),
 			array( 'header-c', 'testc' ),
 		);
-		$request  = $this->mock_request(
-			array(
-				'headers' => $expected,
-			)
-		);
-		$runtime  = new Waf_Runtime( new Waf_Transforms(), new Waf_Operators(), $request );
-		yield 'REQUEST_HEADERS' => array( $runtime, 'request_headers', $expected, '/-b$/' );
+		$runtimeFactory = function ( $that ) use ( $expected ) {
+			$request = $that->mock_request(
```

Comment:
I was trying to find out if we could have done this with a static call instead, but couldn't. It doesn't matter that much, and this works, so why not :)